core/src/test/java/org/apache/iceberg/TestMergeAppend.java

```diff
@@ -373,10 +374,10 @@ public void testChangedPartitionSpec() {
         .bucket("id", 4)
         .build();
 
-    // commit the new partition spec to the table manually
-    table.ops().commit(base, base.updatePartitionSpec(newSpec));
+    // commit the new partition spec to the table
+    table.updatePartitionSpec().update(newSpec).commit();
```

Comment:
As I mentioned earlier, I think this should continue using the `updatePartitionSpec` method so that we can add a public API for this in a separate commit.

Reply:
👌